### DISCUSSION TITLE
Support for open-telemetry

### DISCUSSION THREAD

**Discussion by amitpareek2** [go-resty/resty#648] [General]
Hi Team ,

Is there an implementation provided to support Open telemetry traces ?
Below are the list of sdk supported for opentelemetry https://github.com/open-telemetry/opentelemetry-go-contrib/tree/main/instrumentation where Resty name is missing .

Thanks,
Amit

**Comment by mjlzz** [comment]
you can try with otelhttp
```
client := resty.NewWithClient(
	&http.Client{
		Transport: otelhttp.NewTransport(http.DefaultTransport),
	}
),
```
after init trace provider, you can use client which has otelhttp transport to send request
if the request need to inherit upstream trace, don't forget to set request context `client.R().SetContext(ctx)`
at the server side, you can see the trace info in request header like `Traceparent:[00-3a9714258fe613baf636f830ab967c0a-4498ad5d080d6dbf-01]`

**Reply by FarmerChillax** [reply]
For existing applications, this huge transformation cost.
If resty can use `http.DefaultTransport`, can modify it uniformly

like this: `http.DefaultTransport = otelhttp.NewTransport(http.DefaultTransport)`

**Comment by jeevatkm** [comment]
Please go with @mjlzz suggestion.
